tools/pylib/_boutcore_build/CMakeLists.txt

```
@@ -0,0 +1,135 @@
+option(BOUT_PYTHON "Build the Python interface" MAYBE)
+
+if (BOUT_PYTHON)
+  find_package(Numpy)
+
+  if (NOT ${NUMPY_FOUND})
+    set(MSG "NumPy is required for the python interface!")
+    if ("${BOUT_PYTHON}" STREQUAL "ON")
+      message(FATAL_ERROR ${MSG})
+    else()
+      message(WARNING ${MSG})
+      set(BOUT_PYTHON OFF)
+    endif()
+  endif()
+
+  find_package(Cython)
+  if (NOT ${CYTHON_FOUND})
+    set(MSG "Cython is required for the python interface!")
+    if ("${BOUT_PYTHON}" STREQUAL "ON")
+      message(FATAL_ERROR ${MSG})
+    else()
+      message(WARNING ${MSG})
+      set(BOUT_PYTHON OFF)
+    endif()
+  endif()
+
+  execute_process(COMMAND bash --version
+    RESULT_VARIABLE BASH_FOUND
+    OUTPUT_VARIABLE BASH_VERSION_INFO
+    TIMEOUT 1
+    )
+  if (NOT ${BASH_FOUND} EQUAL 0)
+    set(MSG "bash is required for the python interface!")
+    if ("${BOUT_PYTHON}" STREQUAL "ON")
+      message(FATAL_ERROR ${MSG})
+    else()
+      message(WARNING ${MSG})
+      set(BOUT_PYTHON OFF)
+    endif()
+  endif()
+
+  execute_process(COMMAND ${Python_EXECUTABLE} -c "import sysconfig; print(sysconfig.get_config_var('EXT_SUFFIX'))"
+    RESULT_VARIABLE PYTHON_WORKING
+    OUTPUT_VARIABLE PYTHON_EXT_SUFFIX
+    )
+  if (NOT ${PYTHON_WORKING} EQUAL 0)
+    set(MSG "Failed to get the extension name from python!")
+    if ("${BOUT_PYTHON}" STREQUAL "ON")
+      message(FATAL_ERROR ${MSG})
+    else()
+      message(WARNING ${MSG})
+      set(BOUT_PYTHON OFF)
+    endif()
+  endif()
+  string(STRIP ${PYTHON_EXT_SUFFIX} PYTHON_EXT_SUFFIX)
+
+  if (NOT BUILD_SHARED_LIBS)
+    set(MSG "Need to build as shared lib for the python interface!")
+    if ("${BOUT_PYTHON}" STREQUAL "ON")
+      message(FATAL_ERROR ${MSG})
+    else()
+      message(WARNING ${MSG})
+      set(BOUT_PYTHON OFF)
+    endif()
+  endif()
+
+  # No errors? We can build the interface!
+  if ("${BOUT_PYTHON}" STREQUAL "MAYBE")
+    set(BOUT_PYTHON ON)
+  endif()
+
+
+  if (NOT "${BOUT_PYTHON}" STREQUAL "ON")
+    message(WARNING "Python interface will not be build, see warnings above")
+  else()
```

Could also return here, and then less indentation below?

```suggestion
    message(WARNING "Python interface will not be build, see warnings above")
    return()
  endif()
```